### PR TITLE
slamdata: init at 3.0.11

### PR DESCRIPTION
**Pull request by abuibrahim**
###### Motivation for this change

Add [SlamData](http://slamdata.com)
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/web-apps/slamdata.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  config = mkIf cfg.enable {
+    systemd.services.slamdata = {
+      description = "SlamData";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      preStart = ''
+        mkdir -p "${stateDir}"
+        [ -f "${confFile}" ] || cp "${confFile0}" "${confFile}"
```
Comment (grahamc) [inline nixos/modules/services/web-apps/slamdata.nix]: Why copy the confFile fromconfFile0 to the statedir? More importantly, why only do it once? This means changing the config will never work. Does Quasar alter this config file?

Comment (abuibrahim) [inline nixos/modules/services/web-apps/slamdata.nix]: Unfortunately it does. As you keep using SlamData and adding new workspaces, quasar saves the changes into the config file. I tried to go with the immutable config but quasar loudly complains on a first attempt to modify it.

**Comment by abuibrahim** [comment]
@jdegoes might find this interesting

**Comment by jdegoes** [comment]
@abuibrahim Nice work! Happy to get this on Nix. 🎉

**Comment by joachifm** [comment]
:-1:  from me as is, adding +11,000 lines for a single package is too much.

**Comment by abuibrahim** [comment]
Fair enough. Let me see if I can I trim down some node package dependencies.

**Pull request closed without merging**